// material-ui-filter/es/components/OperatorField.js
import * as filterSelectors from '../store/selectors';
import Delete from '@material-ui/icons/Delete';
import IconButton from '@material-ui/core/IconButton';
import Input from '@material-ui/core/Input';
import React, { Component } from 'react';
import Toolbar from '@material-ui/core/Toolbar';
import Tooltip from '@material-ui/core/Tooltip';
import { SelectField } from 'muishift';
import { withTheme, withStyles } from '@material-ui/core/styles';

var styles = {};

export var OperatorField = function OperatorField(_ref) {
  var queryIndex = _ref.queryIndex,
      currentField = _ref.currentField,
      query = _ref.query,
      fields = _ref.fields,
      operators = _ref.operators,
      handleQueryChange = _ref.handleQueryChange,
      formatMessage = _ref.formatMessage,
      classes = _ref.classes,
      onClick = _ref.onClick;

  var getFieldType = function getFieldType(currentField) {
    var fieldType = '';

    fields.map(function (field) {
      if (field.name === currentField.value) {
        fieldType = field.type;
      }
      return field;
    });

    return fieldType;
  };

  var _filterSelectors$sele = filterSelectors.selectQueryProps(query),
      operator = _filterSelectors$sele.operator,
      isCaseSensitive = _filterSelectors$sele.isCaseSensitive;

  if (queryIndex == null || currentField == null || query == null || handleQueryChange == null || fields == null) {
    return React.createElement('div', null);
  }

  var fieldType = getFieldType(currentField);
  var divFields = [];

  operators.map(function (operator) {
    if (operator.type === fieldType || operator.type === 'string' && fieldType === undefined) {
      operator.operators.map(function (op) {
        return divFields.push({
          value: op.value,
          label: op.label
        });
      });
    }
    return divFields;
  });

  return React.createElement(
    Toolbar,
    null,
    React.createElement(
      'div',
      { style: { maxWidth: 160 } },
      React.createElement(SelectField, {
        input: { value: operator },
        onChange: function onChange(val) {
          handleQueryChange(queryIndex, 'operator', val);
        },
        items: divFields,
        itemToString: function itemToString(item) {
          return item ? item.label : '';
        },
        id: 'react-select-single',
        inputProps: {
          fullWidth: true,
          placeholder: formatMessage ? formatMessage({ id: 'hint_autocomplete' }) : 'Select operator'
        }
      })
    ),
    React.createElement(
      Tooltip,
      {
        id: 'tooltip-bottom-start',
        title: formatMessage ? formatMessage({ id: isCaseSensitive ? 'disable_case_sensitivity' : 'enable_case_sensitivity' }) : '',
        placement: 'bottom-end'
      },
      React.createElement(
        IconButton,
        { onClick: onClick, 'aria-label': 'Delete', color: 'secondary' },
        React.createElement(Delete, null)
      )
    )
  );
};

export default withStyles(styles)(OperatorField);
